internal/authentications/authentication_header.go

```diff
@@ -0,0 +1,65 @@
+package authentications
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/ydataai/authentication-service/internal/configurations"
+	"github.com/ydataai/authentication-service/internal/services"
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// AuthenticationHeader defines a authentication header struct.
+type AuthenticationHeader struct {
+	oidcService    services.OIDCService
+	restCtrlConfig configurations.RESTControllerConfiguration
+	logger         logging.Logger
+}
+
+// NewAuthenticationHeader defines a new AuthenticationHeader struct.
+func NewAuthenticationHeader(logger logging.Logger,
+	oidcService services.OIDCService,
+	restCtrlConfig configurations.RESTControllerConfiguration) CredentialsHandler {
+
+	return &AuthenticationHeader{
+		oidcService:    oidcService,
+		restCtrlConfig: restCtrlConfig,
+		logger:         logger,
+	}
+}
+
+// Extract is an interface that provides authentication from the header.
+func (ah *AuthenticationHeader) Extract(r *http.Request) (map[string]interface{}, error) {
+	// Try to get session from header
+	token := getBearerToken(r.Header.Get(ah.restCtrlConfig.AuthHeader))
+	if token == "" {
+		ah.logger.Infof("%s header not found", ah.restCtrlConfig.AuthHeader)
+
+		// currently, browsers use {Accept: text/html} header.
+		// by the way, if the request comes from a browser,
+		// it probably won't have an Authorization Header, so we should ignore it.
+		if strings.Contains(r.Header.Get("Accept"), "text/html") {
+			return nil, errors.New(ah.restCtrlConfig.AuthHeader + " header not found")
+		}
+	}
+
+	ah.logger.Infof("%s header found: %s", ah.restCtrlConfig.AuthHeader, token)
```

Comment:
What's happen if the `token` is empty and the header `Accept` is `application/json`? According to the logic in place will be success and the log will show `header found`.